crates/trigger/src/cli.rs

```diff
@@ -312,7 +312,7 @@ impl<T: Trigger> TriggerAppBuilder<T> {
         let use_gpu = true;
         let state_dir = match options.state_dir {
             // Make sure `--state-dir=""` unsets the state dir
-            Some(s) if s.trim().is_empty() => None,
+            Some("") => None,
```

Comment:
Sneaky lint fix...

Reply:
This isn't equivalent, no? `Some("  ")` would match the first but not the second.

Reply:
Ah sorry I blindly obeyed clippy. That said, `--state-dir " "` seems more likely to be an accident...

Reply:
Though it now occurs to me that its perfectly valid to create a directory named `" "`...